### PR TITLE
Feature/add tool validation

### PR DESCRIPTION
**Pull request by garyluu**
- Adding more validations for private registries
- Change the tool's title from path to tool_path to handle cases where multiple tools have the same path

### CODE REVIEW

File: src/app/container/register-tool/register-tool.service.ts [hunk 1]
```diff
@@ -59,11 +59,23 @@ export class RegisterToolService {
...
 
-    isInvalidPrivateTool() {
-        return false;
+    isInvalidPrivateTool(toolObj: Tool) {
+        return toolObj.private_access === true && !toolObj.tool_maintainer_email;
     }
-    isInvalidCustomRegistry() {
-        return false;
+    isInvalidCustomRegistry(toolObj: Tool, customDockerRegistryPath: string) {
+            for (let i = 0; i < this.dockerRegistryMap.length; i++) {
```
Comment (denis-yuen) [inline src/app/container/register-tool/register-tool.service.ts]: please add a comment for the method. 
Not sure what this is checking for

File: src/app/container/register-tool/register-tool.service.ts [hunk 2]
```diff
@@ -29,15 +29,15 @@ export class RegisterToolService {
...
-            'container, please ensure that the container attributes are ' +
-            'valid and the same image has not already been registered.',
-            errorDetails: '[HTTP ' + error.status + '] ' + error.statusText + ': ' +
-            error._body
-        };
-    }
-    this.toolRegisterError.next(mapthing);
+            mapthing = {
+                message: 'The webservice encountered an error trying to create this ' +
+                'container, please ensure that the container attributes are ' +
```
Comment (denis-yuen) [inline src/app/container/register-tool/register-tool.service.ts]: This should say tool now

File: src/app/container/register-tool/register-tool.service.ts (hunk 1 above)
Comment (garyluu) [inline src/app/container/register-tool/register-tool.service.ts]: If the registry is private, it checks if the customDockerRegistryPath is truthy (not null, not undefined, not "", etc)

File: src/app/container/register-tool/register-tool.service.ts (hunk 2 above)
Comment (denis-yuen) [inline src/app/container/register-tool/register-tool.service.ts]: This seems to still apply

**Pull request merged**